packages/mobile/src/networkInfo/saga.ts

```diff
@@ -13,6 +15,11 @@ export function* waitForRehydrate() {
   return
 }
 
+export function* waitForNetwork() {
+  yield waitForGethConnectivity()
+  yield checkWeb3Sync()
```

Comment:
From my understanding of the code, this will make a call to web3 to get the latest block every time, which is a little different than the way the rest of our waitFor functions work. Does this need to get the latest block, or is there a way to just signified we reached latest block once already in the session and then we are good to go

Reply:
This function is not actually executing any work, it's just waiting for web3 to sync (meaning `web3.eth.isSyncing` returns `false`).

It doesn't respect the same pattern as when we wait for geth because I didn't want it to be checking the function `web3.eth.isSyncing` and dispatching transactions. We chatted with @jmrossy it might be a good idea to refactor `waitForGethConnectivity` to respect this pattern as well.

What do you think?

Reply:
I get that it's not sending transactions, but it is still making a web3 call, which to me is like making an API call (will have some cost to communicate), and then also afterwards calls getLatestBlock (https://github.com/celo-org/celo-monorepo/pull/100/files#diff-ba0401a32c25d663ec689871b8ec6d86R90). I guess this isn't explicitly defined, but the way I think of our `waitFor` saga actions is to check some state, if it's the proper state then return, otherwise wait for an action that will set that state to something else, and that's it. So they are basically "free" in terms of cost because they don't do much work except checking state and waiting. This breaks that convention and does some work, so that is just why I am a little hesitant about using that name, and also would be hesitant about using this a lot of places

Reply:
I could implement this using the same 'waitForGethConnectivity' and that would be free, but it'd have to pool web3 on the back everytime. This approach only pools web3 before making a contract call.

Reply:
I think if you are going to poll something, just maybe have one worker that manages the polling, and then everything else can just wait on the state to be updated properly

Reply:
The problem with that is that if you have a store variable set to true when it's ready, I'd never go back to false if you're not pooling all the time and would leave the store in an inconsistent state.